settlement-pipelines/src/reporting.rs

```diff
@@ -145,14 +405,20 @@ impl ErrorHandler {
 }
 
 pub async fn with_reporting<T: PrintReportable>(
-    report_handler: &ReportHandler<T>,
+    report_handler: &mut ReportHandler<T>,
     main_result: anyhow::Result<()>,
 ) -> CliResult {
     // print report in whatever case
     report_handler.print_report().await;
     match main_result {
         // when Ok is returned we consult the reality with report handler
-        Ok(_) => CliResult(report_handler.finalize()),
+        Ok(_) => {
+            // before returning result from finalize make possible to adjust entries
+            report_handler
+                .reportable
+                .transform_on_finalize(&mut report_handler.error_handler.entries);
+            CliResult(report_handler.finalize())
```

Comment:
(non-blocking)
There might be some Rust limits around this or it could be impractical for some reasons, but it would feel natural to me to run `self.reportable.transform_on_finalize(&mut self.error_handler.entries)` (or something like that if possible) as the first thing in `report_handler.finalize()` implementation. That way, the caller wouldn't need to think about not forgetting to call `transform_on_finalize` first, it would be done automatically as part of `finalize` when called.

Reply:
It sounds reasonable. I need to change the code a little bit.